R/groupingsets.R

```diff
@@ -85,6 +135,33 @@ groupingsets.data.table = function(x, j, by, sets, .SDcols, id = FALSE, jj, ...)
     set(empty, j = "grouping", value = integer())
     setcolorder(empty, c("grouping", by, setdiff(names(empty), c("grouping", by))))
   }
+  # Define variables related to label
+  if (!is.null(label)) {
+    total.vars = intersect(by, unlist(lapply(sets, function(u) setdiff(by, u))))
+    if (is.list(label)) {
+      by.vars.not.in.label = setdiff(by, names(label))
+      by.vars.not.in.label.class1 = vapply_1c(x[, by.vars.not.in.label, with=FALSE], function(u) class(u)[1])
```

Comment:
1. please avoid calling `[.data.table` here
2. use 1L rather than 1

Reply:
I've changed this.